packages/bpk-component-calendar/src/BpkCalendarGrid.tsx

```diff
@@ -110,15 +111,28 @@ class BpkCalendarGrid extends Component<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    // We cache expensive calculations (and identities) in state
     this.state = {
       calendarMonthWeeks: getCalendarMonthWeeks(
         props.month,
         props.weekStartsOn,
+        // Do not run expensive date format calculations in the constructor
       ),
     };
   }
 
+  // Defer expensive calculations to this point to improve INP.
+  componentDidMount(): void {
+    deferCallback(() =>
+      this.setState({
+        calendarMonthWeeks: getCalendarMonthWeeks(
+          this.props.month,
+          this.props.weekStartsOn,
+          this.props.formatDateFull,
+        ),
+      }),
+    );
+  }
```

Comment:
In the constructor we get dates and formatted labels for everything but the aria-label. Here we regenerate dates, this time with aria labels, and defer until the thread is idle (typically after render). This change reduces INP by roughly 20%.